http_2.rflx

```diff
@@ -0,0 +1,238 @@
+package HTTP_2 is
+
+   type Payload_Length is mod 2**24;
+
+   type Frame_Type is (
+      DATA => 0,
+      HEADERS => 1,
+      PRIORITY => 2,
+      RST_STREAM => 3,
+      SETTINGS => 4,
+      PUSH_PROMISE => 5,
+      PING => 6,
+      GOAWAY => 7,
+      WINDOW_UPDATE => 8,
+      CONTINUATION => 9
+   ) with Size => 8;
+
+   type Settings_Parameter is (
+      SETTINGS_HEADER_TABLE_SIZE => 16#01#,
+      SETTINGS_ENABLE_PUSH => 16#02#,
+      SETTINGS_MAX_CONCURRENT_STREAMS => 16#03#,
+      SETTINGS_INITIAL_WINDOW_SIZE => 16#04#,
+      SETTINGS_MAX_FRAME_SIZE => 16#05#,
+      SETTINGS_MAX_HEADER_LIST_SIZE => 16#06#
+   ) with Size => 16;
+
+   type Settings_Value_Generic is mod 2**32;
+   type Settings_Value_Enable_Push is range 0 .. 1 with Size => 32;
+   type Settings_Value_Initial_Window_Size is range 0 .. 2**31 - 1 with Size => 32;
+   type Settings_Value_Max_Frame_Size is range 2**14 .. 2**24-1 with Size => 32;
+
+   type Zero_Bit is range 0 .. 0 with Size => 1;
+   type Stream_ID is mod 2**31;
+   type Weight is mod 2**8;
+   type Error_Code is (
+      NO_ERROR => 16#00#,
+      PROTOCOL_ERROR => 16#01#,
+      INTERNAL_ERROR => 16#02#,
+      FLOW_CONTROL_ERROR => 16#03#,
+      SETTINGS_TIMEOUT => 16#04#,
+      STREAM_CLOSED => 16#05#,
+      FRAME_SIZE_ERROR => 16#06#,
+      REFUSED_STREAM => 16#07#,
+      CANCEL => 16#08#,
+      COMPRESSION_ERROR => 16#09#,
+      CONNECT_ERROR => 16#0A#,
+      ENHANCE_YOUR_CALM => 16#0B#,
+      INADEQUATE_SECURITY => 16#0C#,
+      HTTP_1_1_REQUIRED => 16#0D#
+   ) with Size => 32;
+
+   type Window_Size_Increment is range 1 .. 2**31 - 1 with Size => 31;
+   type Pad_Length is mod 2**8;
+
+   type Setting is
+      message
+         Settings_Identifier : Settings_Parameter
+            then Settings_Value_Enable_Push
+               if Settings_Identifier = SETTINGS_ENABLE_PUSH
+            then Settings_Value_Initial_Window_Size
+               if Settings_Identifier = SETTINGS_INITIAL_WINDOW_SIZE
+            then Settings_Value_Max_Frame_Size
+               if Settings_Identifier = SETTINGS_MAX_FRAME_SIZE
+            then Settings_Value_Generic
+               if Settings_Identifier = SETTINGS_HEADER_TABLE_SIZE
+                  or Settings_Identifier = SETTINGS_MAX_CONCURRENT_STREAMS
+                  or Settings_Identifier = SETTINGS_MAX_HEADER_LIST_SIZE;
+         Settings_Value_Enable_Push : Settings_Value_Enable_Push
+            then null;
+         Settings_Value_Initial_Window_Size : Settings_Value_Initial_Window_Size
+            then null;
+         Settings_Value_Max_Frame_Size : Settings_Value_Max_Frame_Size
+            then null;
+         Settings_Value_Generic : Settings_Value_Generic;
+      end message;
+
+   type Settings is array of Setting;
```

Comment:
The `Settings` type leads to a conflict with the enum literal `SETTINGS` in the generated code. Renaming one of the two should fix that problem.

Reply:
Renamed `Setting` to `Settings_Parameter` and the former `Settings_Parameter` to `Settings_Identifier`. 
This naming is consistent with the naming in the RFC.